laws/shared/src/main/scala/algebra/laws/RingLaws.scala

```diff
@@ -57,17 +45,19 @@ trait RingLaws[A] extends GroupLaws[A] {
     parent = Some(multiplicativeMonoid)
   )
 
-  def multiplicativeGroup(implicit A: MultiplicativeGroup[A]) = new MultiplicativeProperties(
-    base = _.group(A.multiplicative),
-    parent = Some(multiplicativeMonoid),
-    // pred is used to ensure y is not zero.
-    "consistent division" -> forAll { (x: A, y: A) =>
-      pred(y) ==> (A.div(x, y) ?== A.times(x, A.reciprocal(y)))
-    }
-  )
+  def multiplicativeGroup(implicit A: MultiplicativeGroup[A] with AdditiveMonoid[A]) = {
+    implicit val Arb: Arbitrary[A] = nonZeroArb(A)
+    new MultiplicativeProperties(
+      base = _.group(A.multiplicative, Arb),
+      parent = Some(multiplicativeMonoid),
+      "consistent division" -> forAll{ (x: A, y: A) =>
+        (A.div(x, y) ?== A.times(x, A.reciprocal(y)))
+      }
+    )
```

Comment:
The non-zero arbitrary `Arb` is implicit here in the local scope only. The only reason it is there implicitly is to use it in the `"consistent division"` property. If you can think of a clear way to pass it to that property explicitly, or to limit the scope of the implicit value even more, I will make a change.